lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -158,6 +162,11 @@ defmodule LambdaEthereumConsensus.ForkChoice do
     {:noreply, new_store}
   end
 
+  @impl GenServer
+  def handle_info(_msg, state) do
```

Comment:
Instead of calling a `cast` to update the genserver, we can match the message against `{_ref, new_fork_choice_cache}` and receive the task result directly.